src/components/UI/Pagination/Pagination.test.tsx

```diff
@@ -0,0 +1,28 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import React from "react";
+import { Pagination } from "./Pagination";
+
+describe("pagination", () => {
+  it("should render pagination component correctly on initial load", () => {
+    render(<Pagination totalNoOfPages={8} currentPage={1} setCurrentPage={() => {}} />);
+
+    expect(screen.getAllByTestId(/page-number/)).toHaveLength(6);
+  });
+
```

Comment:
might need some tests for near edge cases, say:
- when active is at 3, left truncation should not appear
- when active is at 4, left truncation should appear

same goes with the right side

Reply:
ok